src/main/java/Duke.java

```diff
@@ -20,17 +22,27 @@ public static void main(String[] args) {
         System.out.println("\n_______________________________________________");
 
         do{
-            line = in.nextLine();
-            if(!line.equals("bye") && !line.equals("list")){
+            line = in.next();
+            if(!line.equals("bye") && !line.equals("list") && !line.equals("done")){
+                secondline = in.nextLine();
+                line = line + secondline;
+                Task t = new Task(line);
                 System.out.println("added: " + line + "\n");
-                texts[NumberOfTexts] = line;
+                texts[NumberOfTexts] = t;
                 NumberOfTexts++;
             }
             if(line.equals("list")){
+                System.out.println("Here are the tasks in your list:");
                 for(int i = 0; i < NumberOfTexts; i++){
-                    System.out.println(i+1 + ". " + texts[i]);
+                    System.out.println(i+1 + "." + "[" + texts[i].getStatusIcon() + "] " + texts[i] );
                 }
             }
+            if(line.equals("done")){
+                TextNumber = in.nextInt() - 1;
+                texts[TextNumber].markAsDone();
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println("  [" + texts[TextNumber].getStatusIcon() + "] " + texts[TextNumber]);
+            }
 
         }while(!line.equals("bye"));
 
```

Comment:
Perhaps this new line is unnecessary, you may want to remove this so your code looks slightly neater.